### DISCUSSION TITLE
Help troubleshooting fzf-lua grep

### DISCUSSION THREAD

**Discussion by dotfrag** [LazyVim/LazyVim#3656] [Q&A] [answered]
Hello, I'm having a problem with LazyVim and fzf-lua grep that I do not know how to troubleshoot.

Using fzf-lua [mini script](https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#quickstart), grep works as expected:

![2024-06-14T22:55:30,645113542+03:00](https://github.com/LazyVim/LazyVim/assets/17456867/9e85c885-1ca0-4be8-ab61-5cd53a6e43fd)

However, using LazyVim's `repro.lua` and enabling the fzf extra, I get this:

![2024-06-14T22:57:04,741053239+03:00](https://github.com/LazyVim/LazyVim/assets/17456867/5bd9abc6-0572-4b49-83a3-11f3e2c11dd9)

If I zoom out, I see this:

![2024-06-14_22-58](https://github.com/LazyVim/LazyVim/assets/17456867/a967457c-df2b-448e-8bb4-ee287f447aa4)

I would ask in `fzf-lua` repo but I'm not having the problem there. Any help is appreciated.

**Comment by dpetka2001** [comment]
Why are you using the `repro.lua`? This is just for submitting bug issues. Do you have the same problem in your personal configuration?

**Reply by dotfrag** [reply]
Yes I do have the same problem in my personal config. I used `repro.lua` to eliminate the case of me screwing up in the config.

**Reply by dpetka2001** [reply]
Did you enable the fzf Extra via `LazyExtras`?

**Reply by dotfrag** [reply]
Yes:

```json
{
  "extras": [
    "lazyvim.plugins.extras.editor.fzf"
  ],
  "news": {
    "NEWS.md": "6077"
  },
  "version": 6
}
```

**Reply by dpetka2001** [reply]
Can you try the following
```bash
1. `mkdir ~/.config/lazytest`
2. `git clone https://github.com/LazyVim/starter ~/.config/lazytest`
3. `NVIM_APPNAME=lazytest nvim` and wait for the installation to complete
4. While Lazy UI is open, press `C` to check for updates and then `U` to update LazyVim to latest version.
5. Restart Neovim with command from step 3 and enable fzf-lua Extra from `LazyExtras` UI and restart again
```
Do you observe the same behavior? 

For cleanup
```bash
rm -rf ~/.config/lazytest
rm -rf ~/.local/share/lazytest
rm -rf ~/.local/state/lazytest
rm -rf ~/.cache/lazytest
```

**Reply by dotfrag** [reply]
Same behaviour.

```text
nvim-lazyvim on  main [?] via 🌙 v5.4.6 
❯ git remote -v
origin	https://github.com/LazyVim/starter (fetch)
origin	https://github.com/LazyVim/starter (push)

nvim-lazyvim on  main [?] via 🌙 v5.4.6 
❯ cat lazyvim.json 
   1   │ {
   2   │   "extras": [
   3   │     "lazyvim.plugins.extras.editor.fzf"
   4   │   ],
   5   │   "news": {
   6   │     "NEWS.md": "6077"
   7   │   },
   8   │   "version": 6
   9   │ }
```

**Reply by dpetka2001** [reply]
I can't reproduce. Both `repro.lua` and LazyVim default starter configuration work as they should. Where are you installing Neovim? What is that path `file://xps/tmp/tmp.xxxx/repro.lua`

**Reply by dpetka2001** [reply]
The steps that I showed you, did actually clone the repo in `~/.config/lazytest/` directory? Or did you clone it at a different location?

**Reply by dotfrag** [reply]
I used the same commands, it was cloned to a different directory just because I already have a script that handles this for me that I use to test different configs.

The path you asked from the initial screenshots is the URI of the file I had in the cwd. `file://` is the "protocol", `xps` the hostname, `/tmp/..` actual filepath. For example, if I `cd ~/.config/nvim-lazyvim`, I get `file://xps/home/frag/.config/nvim-lazyvim/init...`

What is also weird, is that I get the exact same behaviour on two very different machines. One is running arch linux and the other ubuntu.

**Reply by dpetka2001** [reply]
If it's not cloned in `~/.config/lazytest` then `NVIM_APPNAME` won't work and you're essentially testing your  own config. That won't work. Is that script you're using symlinking the directories to somewhere else? Can you try without using that script? Just clone it yourself manually into `~/.config/lazytest/` directory and check behavior.

**Reply by dotfrag** [reply]
I'm doing the same thing as you posted, just modified the commands. https://github.com/dotfrag/dotfiles/blob/main/bin/nvims.

Anyhow, just to be 150% I'm not doing anything stupid, same behaviour with the exact commands you sent:

![2024-06-14T23:54:31,491020477+03:00](https://github.com/LazyVim/LazyVim/assets/17456867/24db23f1-0d99-49f8-9900-7b6a41a3afe4)

**Reply by dpetka2001** [reply]
Can you try the following command `FzfLua live_grep formatter="path.filename_first"`? How does that behave?

**Comment by dotfrag** [comment, marked as answer]
I found the problem. I have the env var `RIPGREP_CONFIG_PATH` set in my shell pointing to `~/.config/ripgrep/config`. The contents of the file are:

```text
--max-columns=150
--max-columns-preview
--smart-case
--hyperlink-format=default
```

The line `--hyperlink-format=default` was causing the problem because it modifies the output with hyperlinks, which is why I was seeing the `file://` part in the output.

Thanks @dpetka2001 for the help, the troubleshooting made me realise the problem was with my environment setup.

**Reply by dpetka2001** [reply]
I didn't really do anything, but I'm glad you got to the end of it. Also thanks for sharing your solution, it might prove useful for other users as well :smile:

**Comment by dotfrag** [comment]
@folke if you are interested in solving this problem in LazyVim, this option needs to be passed to rg options `--hyperlink-format=none` to override user config. 

`man rg`:

> A format string may be empty. An empty format string is equivalent to the none alias. In this case, hyperlinks will be disabled.

**Reply by folke** [reply]
See https://github.com/ibhagwan/fzf-lua/issues/1266

Would be good to be able to (optionally) reset the env used by fzf-lua
